thrift/src/test/java/com/linecorp/armeria/common/thrift/AsyncMethodCallbacksTest.java

```diff
@@ -0,0 +1,110 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.common.thrift;
+
+import static com.spotify.futures.CompletableFutures.exceptionallyCompletedFuture;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.argThat;
+import static org.mockito.ArgumentMatchers.isA;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import org.apache.thrift.async.AsyncMethodCallback;
+import org.junit.Test;
+
+import com.linecorp.armeria.testing.internal.AnticipatedException;
+
+public class AsyncMethodCallbacksTest {
+
+    @Test
+    public void transferSuccess() {
+        @SuppressWarnings("unchecked")
+        final AsyncMethodCallback<String> callback = mock(AsyncMethodCallback.class);
+        AsyncMethodCallbacks.transfer(completedFuture("foo"), callback);
+
+        verify(callback, times(1)).onComplete("foo");
+        verify(callback, never()).onError(any());
+    }
+
+    @Test
+    public void transferFailure_Exception() {
+        @SuppressWarnings("unchecked")
+        final AsyncMethodCallback<String> callback = mock(AsyncMethodCallback.class);
+        AsyncMethodCallbacks.transfer(exceptionallyCompletedFuture(new AnticipatedException()), callback);
+
+        verify(callback, never()).onComplete(any());
+        verify(callback, times(1)).onError(isA(AnticipatedException.class));
+    }
+
+    @Test
+    public void transferFailure_Throwable() {
+        @SuppressWarnings("unchecked")
+        final AsyncMethodCallback<String> callback = mock(AsyncMethodCallback.class);
+        AsyncMethodCallbacks.transfer(exceptionallyCompletedFuture(new Throwable("foo")), callback);
+
+        verify(callback, never()).onComplete(any());
+        verify(callback, times(1)).onError(argThat(argument -> {
+            return argument instanceof CompletionException &&
+                   "foo".equals(argument.getCause().getMessage());
+        }));
+    }
+
+    @Test
+    public void transferCallbackError() {
+        @SuppressWarnings("unchecked")
+        final AsyncMethodCallback<String> callback = mock(AsyncMethodCallback.class);
+        doThrow(new AnticipatedException()).when(callback).onComplete(any());
+        AsyncMethodCallbacks.transfer(completedFuture("foo"), callback);
```

Comment:
Is it better to add assertions here?
`verify(callback, only()).onComplete("foo");`

Reply:
Sounds good. Added a verification as suggested.